### DISCUSSION TITLE
Permission to use uazo's Android autofill patch under MIT licence

### DISCUSSION THREAD

**Discussion by ghost** [bromite/bromite#1571] [Builds]
Hello, I would contact @uazo personally but they do not have any public email so I'll ask here.

I was wondering if you'd allow your patch to be used for GrapheneOS's Vanadium (The Bromite developers are aware of it) under the MIT licence. As far as I'm aware all the patches in this repository, unless specified, are GPLv3.

The patch that would like to be used is: https://github.com/bromite/bromite/blob/master/build/patches/Enable-native-Android-autofill.patch

The project's repo: https://github.com/GrapheneOS/Vanadium

**Comment by uazo** [comment]
> I was wondering if you'd allow your patch to be used for GrapheneOS's Vanadium 

thanks for the request, very professional.
absolutely yes, for me there is no problem, indeed, I am pleased
also consider that there is little of mine, the code was already present in the chromium repo, I just connected it.

**Reply by uazo** [reply]
ah, just one thing.
the patch is actually incomplete. I realized after that it is also active for incognito browsing without user authorization. it is something that I will solve soon, indeed, if you want to do it...

**Reply by ghost** [reply]
Thank you very much. I'm no licencing expert and I was told I needed to ask _because_ of licencing.

I haven't looked at any other autofill patches that might have been in the repo for native autofill; I just saw that first one that implements it and assumed that's all was necessary. Are there any other patches that would be needed? We're likely willing to fix up the patch that you mentioned too. I'm also not a Chromium expert so I wouldn't know much about this stuff, haha. I'll let some people know though.

Thank you so much again. A lot of GrapheneOS users would like native autofill (me!) but they have to resort to accessibility services, something GrapheneOS frowns upon and makes note of it in Auditor (Attestation).

**Reply by quh4gko8** [reply]
Thanks for allowing this patch usage for Vanadium. 

I'm curious though, why are the arguments here:
https://github.com/bromite/bromite/blob/master/build/patches/Enable-native-Android-autofill.patch#L848,L849

are the other way around (the last two lines), compared to the other lines changed using that function?

See: https://github.com/bromite/bromite/blob/master/build/patches/Enable-native-Android-autofill.patch#L359,L366

**Reply by quh4gko8** [reply]
Also, is there any particular reason 2 DCHECKS are used here?

https://github.com/bromite/bromite/blob/master/build/patches/Enable-native-Android-autofill.patch#L353,L355

Here is the patch ported to Vanadium:
https://github.com/repository-staging/Vanadium/commit/10944776f27236a8ffb2cfbafec5e11147f7ce32#diff-afceca676dcf548bc38ceed8ee633529b7fd92537c56cf65016d4836ad5a3df8

modified to fix some build errors, such as not finding the `ContentViewWithAutofill` constructor, and the comment above

**Reply by ghost** [reply]
@quh4gko8 I would ping them and ask these questions in https://github.com/GrapheneOS/Vanadium/issues/132

Unless @uazo responds to you here.

**Comment by csagan5** [comment]
> Hello, I would contact @uazo personally but they do not have any public email so I'll ask here.
> 
> I was wondering if you'd allow your patch to be used for GrapheneOS's Vanadium (The Bromite developers are aware of it) under the MIT licence. As far as I'm aware all the patches in this repository, unless specified, are GPLv3.

I think it's fine, in this case the author can re-license it; GPLv3 is in place because if someone creates a fork and fills it with malware (or makes it send searches to a third-party website, already happened) then suddenly the patch authors might not like to be associated with such work. But the license itself does not guard against this, it is only a safety measure so that further modifications are open sourced and more transparent.

**Reply by ghost** [reply]
Thanks a lot for the clarification and maintainer approval of my request! I'm no licencing expert so appreciate the tip.
